### PR TITLE
MM-15566 Adding new CI

### PR DESCRIPTION
**Pull request by crspeller**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,81 +1,58 @@
 version: 2.1
-executors:
-  default:
-    working_directory: /go/src/github.com/mattermost/mattermost-plugin-github
-    docker:
-      - image: circleci/golang:1.12-node
 
+orbs:
+  plugin-ci: mattermost/plugin-ci@volatile
```
Comment (hanzei) [inline .circleci/config.yml]: Out of interest: Is there a repo for this?

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,20 @@
+module github.com/mattermost/mattermost-plugin-github
+
+go 1.12
+
+require (
+	github.com/go-ldap/ldap v3.0.3+incompatible // indirect
+	github.com/google/go-github v17.0.0+incompatible
```
Comment (hanzei) [inline go.mod]: Maybe this is a good time to update this? v17.0.0 is a year old.

Comment (cpanato) [reply go.mod]: If we update that we need to change some code there are some breaking changes

Comment (cpanato) [reply go.mod]: However if need i can do the update

Comment (hanzei) [reply go.mod]: I'm 1/5 that this dependency should be updated at some point. 0/5 if this PR is the right place.

File: .circleci/config.yml (hunk 1 above)
Comment (crspeller) [reply .circleci/config.yml]: Soon.

File: go.mod (hunk 2 above)
Comment (crspeller) [reply go.mod]: Another PR.

File: .circleci/config.yml (hunk 1 above)
Comment (crspeller) [reply .circleci/config.yml]: https://github.com/mattermost/circleci-orbs

**Pull request merged**